### PR TITLE
Adding creationTime and expirationTime properties to TableListItem

### LINKED ISSUE #5117: BigQuery: Add creationTime and expirationTime properties to TableListItem
API reference: https://cloud.google.com/bigquery/docs/reference/rest/v2/tables/list

The properties `creationTime` and `expirationTime` now appear in the API results for `list_tables()`. The `creationTime` properties is exposed as `Table.created`. A similar property needs to appear on `TableListItem`.

https://github.com/GoogleCloudPlatform/google-cloud-python/blob/62740078538ea84167705c15bf109c1a2737c648/bigquery/google/cloud/bigquery/table.py#L387-L397

Likewise `Table.expires` needs a corresponding property.

https://github.com/GoogleCloudPlatform/google-cloud-python/blob/62740078538ea84167705c15bf109c1a2737c648/bigquery/google/cloud/bigquery/table.py#L546-L556

### PR DESCRIPTION
**Pull request by lbristol88**
Closes #5117 

@tswast @engelke

### CODE REVIEW

File: bigquery/tests/unit/test_table.py [hunk 1]
```diff
@@ -1138,6 +1147,17 @@ def test_ctor(self):
         }
 
         table = self._make_one(resource)
+
+        if "creationTime" in resource:
```
Comment (tswast) [inline bigquery/tests/unit/test_table.py]: `creationTime` is in resource, as you are setting it above. This if statement is not needed. Just assert on the expected value. Same applies for `expirationTime`.

File: bigquery/tests/unit/test_table.py [hunk 2]
```diff
@@ -1115,11 +1115,20 @@ def _get_target_class():
     def _make_one(self, *args, **kw):
         return self._get_target_class()(*args, **kw)
 
+    def _setUpConstants(self):
+        import datetime
+        from google.cloud._helpers import UTC
+
+        self.WHEN_TS = 1437767599.006
```
Comment (tswast) [inline bigquery/tests/unit/test_table.py]: `.006` is not accurately representable in base 2, so you may encounter rounding issues with this value. Choose a fraction such as `.125` instead.

File: bigquery/google/cloud/bigquery/table.py [hunk 3]
```diff
@@ -906,6 +906,42 @@ def __init__(self, resource):
...
+            ValueError: For invalid value types.
+        """
+        expiration_time = self._properties.get("expirationTime")
+        if expiration_time is not None:
+            # expiration_time will be in milliseconds.
+            return google.cloud._helpers._datetime_from_microseconds(
+                1000.0 * float(expiration_time)
+            )
+
+    @expires.setter
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/table.py]: TableListItem is read-only, so a setter is not needed.

File: bigquery/google/cloud/bigquery/table.py [hunk 4]
```diff
@@ -906,6 +906,42 @@ def __init__(self, resource):
...
+            return google.cloud._helpers._datetime_from_microseconds(
+                1000.0 * float(creation_time)
+            )
+
+    @property
+    def expires(self):
+        """Union[datetime.datetime, None]: Datetime at which the table will be
+        deleted.
+
+        Raises:
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/table.py]: The `Raises` section can be removed when the setter is removed.

File: bigquery/tests/unit/test_table.py (hunk 1 above)
Comment (lbristol88) [reply bigquery/tests/unit/test_table.py]: Removed the If statements for creation an expiration, added asserts on expected value.

File: bigquery/tests/unit/test_table.py (hunk 2 above)
Comment (lbristol88) [reply bigquery/tests/unit/test_table.py]: Made change to .125 as recommended.

File: bigquery/google/cloud/bigquery/table.py (hunk 3 above)
Comment (lbristol88) [reply bigquery/google/cloud/bigquery/table.py]: Removed the setter.

File: bigquery/google/cloud/bigquery/table.py (hunk 4 above)
Comment (lbristol88) [reply bigquery/google/cloud/bigquery/table.py]: Removed the raises comment since setter was removed!

File: bigquery/tests/unit/test_table.py [hunk 5]
```diff
@@ -1174,7 +1170,7 @@ def test_ctor(self):
         self.assertEqual(table.time_partitioning.field, "mycolumn")
         self.assertEqual(table.labels["some-stuff"], "this-is-a-label")
         self.assertIsNone(table.view_use_legacy_sql)
-        self.assertIsNone(table.expires)
+        # self.assertIsNone(table.expires)
```
Comment (tswast) [inline bigquery/tests/unit/test_table.py]: Remove, don't comment out.

Actually, could you move this `assertIsNone` for `expires` and add one for  `created` to `test_ctor_missing_properties`?

Comment (tswast) [reply bigquery/tests/unit/test_table.py]: Thanks for removing. Could you add assertions for `None` values to the `test_ctor_missing_properties` test, too?

Comment (lbristol88) [reply bigquery/tests/unit/test_table.py]: Added assertions for created and expires under the `test_ctor_missing_properties` test.

**Pull request merged**